unit-tests/src/main/scala/java/net/InetAddressSuite.scala

```diff
@@ -0,0 +1,158 @@
+    assertEquals("0:0:0:0:0:0:0:1",
+                 InetAddress.getByName("::1").getHostAddress())
+  }
+
+  test("isReachable") {
+    // Linux disables ICMP requests by default and most of the addresses
+    // don't have echo servers running on port 7, so it's quite difficult
+    // to test this method here
+
+    val addr2 = InetAddress.getByName("127.0.0.1")
+    assertThrows[IllegalArgumentException] { addr2.isReachable(-1) }
+  }
+
+  test("isMulticastAddress") {
+    val ia1 = InetAddress.getByName("239.255.255.255")
+    assert(ia1.isMulticastAddress())
+    val ia2 = InetAddress.getByName("localhost")
+    assertNot(ia2.isMulticastAddress())
+  }
+
+  test("isAnyLocalAddress") {
+    val ia1 = InetAddress.getByName("239.255.255.255")
+    assertNot(ia1.isAnyLocalAddress())
+    val ia2 = InetAddress.getByName("localhost")
+    assertNot(ia2.isAnyLocalAddress())
+  }
+
+  test("isLinkLocalAddress") {
+    val ia1 = InetAddress.getByName("239.255.255.255")
+    assertNot(ia1.isLinkLocalAddress())
+    val ia2 = InetAddress.getByName("localhost")
+    assertNot(ia2.isLinkLocalAddress())
+  }
+
+  test("isLoopbackAddress") {
+    val ia1 = InetAddress.getByName("239.255.255.255")
+    assertNot(ia1.isLoopbackAddress())
+    val ia2 = InetAddress.getByName("localhost")
+    assert(ia2.isLoopbackAddress())
+    val ia3 = InetAddress.getByName("127.0.0.2")
+    assert(ia3.isLoopbackAddress())
+  }
+
+  test("isSiteLocalAddress") {
+    val ia1 = InetAddress.getByName("239.255.255.255")
+    assertNot(ia1.isSiteLocalAddress())
+    val ia2 = InetAddress.getByName("localhost")
+    assertNot(ia2.isSiteLocalAddress())
+    val ia3 = InetAddress.getByName("127.0.0.2")
+    assertNot(ia3.isSiteLocalAddress())
+    val ia4 = InetAddress.getByName("243.243.45.3")
+    assertNot(ia4.isSiteLocalAddress())
+    val ia5 = InetAddress.getByName("10.0.0.2")
+    assert(ia5.isSiteLocalAddress())
+  }
+
+  test("MC methods") {
+    val ia1 = InetAddress.getByName("239.255.255.255")
+    assertNot(ia1.isMCGlobal())
+    assertNot(ia1.isMCLinkLocal())
+    assertNot(ia1.isMCNodeLocal())
+    assertNot(ia1.isMCOrgLocal())
+    assert(ia1.isMCSiteLocal())
+
+    val ia2 = InetAddress.getByName("243.243.45.3")
+    assertNot(ia2.isMCGlobal())
+    assertNot(ia2.isMCLinkLocal())
+    assertNot(ia2.isMCNodeLocal())
+    assertNot(ia2.isMCOrgLocal())
+    assertNot(ia2.isMCSiteLocal())
+
+    val ia3 = InetAddress.getByName("250.255.255.254")
+    assertNot(ia3.isMCGlobal())
+    assertNot(ia3.isMCLinkLocal())
+    assertNot(ia3.isMCNodeLocal())
+    assertNot(ia3.isMCOrgLocal())
+    assertNot(ia3.isMCSiteLocal())
+
+    val ia4 = InetAddress.getByName("10.0.0.2")
+    assertNot(ia4.isMCGlobal())
+    assertNot(ia4.isMCLinkLocal())
+    assertNot(ia4.isMCNodeLocal())
+    assertNot(ia4.isMCOrgLocal())
+    assertNot(ia4.isMCSiteLocal())
+  }
+
+  test("toString") {
+    assertEquals("/127.0.0.1", InetAddress.getByName("127.0.0.1").toString)
+  }
+
+}
```

Comment:
Most of the tests are really similar to the ones in InetAddressTest, I will see how much of them don't overlap.